connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -505,7 +506,9 @@ private SinkRecord convertAndTransformRecord(final ConsumerRecord<byte[], byte[]
                 headers);
         log.trace("{} Applying transformations to record in topic '{}' partition {} at offset {} and timestamp {} with key {} and value {}",
                 this, msg.topic(), msg.partition(), msg.offset(), timestamp, keyAndSchema.value(), valueAndSchema.value());
-        recordActiveTopic(origRecord.topic());
+        if (workerConfig.getBoolean(TOPIC_TRACKING_ENABLE_CONFIG)) {
```

Comment:
`workerConfig.getBoolean(String)` is cached, but there's also the extra logic of tracking that the property was used. Given how this method is called on every record, it seems worth it to speed this up.

How about either pulling this out to a member variable here or in WorkerConfig and adding a getter method to just return the boolean value?

Reply:
Good observation. My thought was that JIT would take care of this pretty soon. But doesn't hurt to add this as a `final` member variable where it's used.